.github/workflows/build-android-app.yml

```
@@ -0,0 +1,144 @@
+          stat ./org.microbiomedata.fieldnotes.keystore.tar.gz.enc
+          echo "Decrypting the keystore file."
+          echo "${{ secrets.ANDROID_KEYSTORE_DECRYPTION_KEY }}" | openssl enc -d -aes256 -pbkdf2 -in ./org.microbiomedata.fieldnotes.keystore.tar.gz.enc | tar xz
+          stat ./org.microbiomedata.fieldnotes.keystore
+      # Reference: https://developer.android.com/studio/publish/app-signing.html#secure-shared-keystore
+      - name: Create `keystore.properties` file
+        working-directory: android
+        run: |
+          echo 'storePassword=${{ secrets.ANDROID_KEYSTORE_PASSWORD }}'   >  ./keystore.properties
+          echo 'keyPassword=${{ secrets.ANDROID_KEYSTORE_KEY_PASSWORD }}' >> ./keystore.properties
+          echo 'keyAlias=${{ secrets.ANDROID_KEYSTORE_KEY_ALIAS }}'       >> ./keystore.properties
+          echo 'storeFile=../org.microbiomedata.fieldnotes.keystore'      >> ./keystore.properties
+          wc -l ./keystore.properties
+      # Add lines that reference the `keystore.properties` file, to the top of the `build.gradle` file.
+      # Note: To effectively "prepend" the lines, we first write everything to a temporary file in the
+      #       order we want, then we overwrite the target file with the contents of that temporary file.
+      # Reference: https://developer.android.com/studio/publish/app-signing.html#secure-shared-keystore
+      - name: Update `app/build.gradle` file to reference `keystore.properties` file
+        working-directory: android
+        run: |
+          echo 'def keystorePropertiesFile = rootProject.file("keystore.properties")' >  ./app/build.gradle.tmp
+          echo 'def keystoreProperties = new Properties()'                            >> ./app/build.gradle.tmp
+          echo 'keystoreProperties.load(new FileInputStream(keystorePropertiesFile))' >> ./app/build.gradle.tmp
+          echo ''                                                                     >> ./app/build.gradle.tmp
+          cat  ./app/build.gradle                                                     >> ./app/build.gradle.tmp
+          mv   ./app/build.gradle.tmp ./app/build.gradle
+      # Reference: https://developer.android.com/studio/publish/app-signing.html#secure-shared-keystore
+      # Reference: https://developer.android.com/build/building-cmdline#gradle_signing
+      - name: Add `signingConfigs` section to `app/build.gradle` file
+        working-directory: android
+        run: |
+          line_number=$(awk '/android *{/{print NR; exit}' app/build.gradle)
+          echo "Inserting text within the 'android' block on line ${line_number}"
+          sed -i "${line_number}a \    signingConfigs {\n        release {\n            keyAlias keystoreProperties['keyAlias']\n            keyPassword keystoreProperties['keyPassword']\n            storeFile file(keystoreProperties['storeFile'])\n            storePassword keystoreProperties['storePassword']\n        }\n    }\n" ./app/build.gradle
```

I suggest reformatting this command so that (a) it doesn't extend so far horizontally, and (b) it more closely visually resembles the content it is adding to the configuration file.

```suggestion
          sed -i "${line_number}a \
              signingConfigs {\n\
                  release {\n\
                      keyAlias keystoreProperties['keyAlias']\n\
                      keyPassword keystoreProperties['keyPassword']\n\
                      storeFile file(keystoreProperties['storeFile'])\n\
                      storePassword keystoreProperties['storePassword']\n\
                  }\n\
              }\n" ./app/build.gradle
```